pkg/controller/admissionchecks/provisioning/controller.go

```diff
@@ -317,12 +288,17 @@ func (c *Controller) syncOwnedProvisionRequest(ctx context.Context, wl *kueue.Wo
 			return nil, err
 		}
 	}
-	return requeAfter, nil
+	return requeueAfter, nil
 }
 
-func (c *Controller) remainingTimeToRetry(pr *autoscaling.ProvisioningRequest, failuresCount int32) time.Duration {
-	backoffDuration := time.Duration(c.minBackoffSeconds) * time.Second
-	maxBackoffDuration := time.Duration(c.maxBackoffSeconds) * time.Second
+func (c *Controller) remainingTimeToRetry(wlInfo *workloadInfo, pr *autoscaling.ProvisioningRequest, failuresCount int32, prc *kueue.ProvisioningRequestConfig) time.Duration {
+	if !features.Enabled(features.KeepQuotaForProvReqRetry) {
+		return wlInfo.requeueState.RequeueAt.Sub(time.Now())
```

Comment:
Hm is this what we want? IIUC RequeueAt should be in the future.
If this is correct then add a comment to explain when / where the backoff is applied in this case.

Reply:
Oh, I misread it, actually you subtract Now which seems ok-ish. 

Still, I'm wondering if we could simply return 0 in this scenario. My thinking is that we only get here if the workload has QuotaReserved=true, and this Requeued=True, and thus RequeueAt should be in the past. 

Possibly I'm missing some scenario above, but it would be good to capture it in a comment in that case.

Reply:
Indeed, thanks for catching that. Done